templates/observability-remote-write-proxy.yml

```diff
@@ -0,0 +1,337 @@
+            resources:
+              requests:
+                cpu: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_CPU_RESOURCE_REQUESTS}}
+                memory: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_MEMORY_RESOURCE_REQUESTS}}
+              limits:
+                cpu: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_CPU_RESOURCE_LIMITS}}
+                memory: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_MEMORY_RESOURCE_LIMITS}}
+            readinessProbe:
+              httpGet:
+                path: /healthcheck
+                port: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}}
+                scheme: HTTP
+                httpHeaders:
+                - name: User-Agent
+                  value: Probe
+              initialDelaySeconds: 5
+              periodSeconds: 10
+            securityContext:
+              allowPrivilegeEscalation: false
+              capabilities:
+                drop:
+                - ALL
+            volumeMounts:
+            - name: observability-remote-write-proxy-oidc-config
+              mountPath: ${OBSERVABILITY_REMOTE_WRITE_PROXY_OIDC_CONFIG_FILEDIR}
+          volumes:
+           - name: observability-remote-write-proxy-oidc-config
+             secret:
+               secretName: observability-remote-write-proxy-oidc-config
+               defaultMode: 288 # 0440 in octal mode permissions
+               optional: true
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      labels:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+      name: observability-rw-proxy
+    spec:
+      ports:
+        - name: http
+          port: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}}
+          targetPort: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}}
+          protocol: TCP
+      selector:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      labels:
+        app.kubernetes.io/component: prometheus-remote-write-proxy-metrics
+        app.kubernetes.io/name: observability-remote-write-proxy-metrics
+      name: observability-rw-proxy-metrics
+    spec:
+      ports:
+        - name: metrics
+          port: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}}
+          targetPort: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}}
+          protocol: TCP
+      selector:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: PodDisruptionBudget
+    apiVersion: policy/v1
+    metadata:
+      name: observability-remote-write-proxy-pdb
+    spec:
+      minAvailable: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PDB_MIN_AVAILABLE}}
+      selector:
+        matchLabels:
+          app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: NetworkPolicy
```

Comment:
@pb82 any thoughts on this?

Reply:
I don't think it's needed and in the future we may want to remove the control plane token refresher in favour of the proxy (although that still needs to be thought out because of external access).

Reply:
I am going to remove it from the PR then.
For this, I am going to add a new comment that removes it without squashing it. The reason for that is so we can see how it was configured in case we need to add it in the future.